core/unit_test/tstParticleList.hpp

```diff
@@ -0,0 +1,227 @@
+
+            for ( int d = 0; d < 3; ++d )
+                get( particle, Cabana::Field::Position<3>(), d ) += p + d;
+
+            get( particle, Foo() ) += p;
+
+            get( particle, Cabana::Field::CommRank() ) += p;
+
+            for ( int i = 0; i < 3; ++i )
+                for ( int j = 0; j < 3; ++j )
+                    get( particle, Bar(), i, j ) += p + i + j;
+
+            aosoa.setTuple( p, particle.tuple() );
+        } );
+
+    // Check the modification.
+    Cabana::deep_copy( aosoa_host, aosoa );
+    for ( std::size_t p = 0; p < num_p; ++p )
+    {
+        for ( int d = 0; d < 3; ++d )
+            EXPECT_DOUBLE_EQ( px_h( p, d ), 1.23 + p + d );
+
+        EXPECT_DOUBLE_EQ( pm_h( p ), 3.3 + p );
+
+        EXPECT_EQ( pc_h( p ), 5 + p );
+
+        for ( int i = 0; i < 3; ++i )
+            for ( int j = 0; j < 3; ++j )
+                EXPECT_DOUBLE_EQ( pf_h( p, i, j ), -1.2 + p + i + j );
+    }
+}
+
+//---------------------------------------------------------------------------//
+void particleViewTest()
+{
+    // Make a particle list.
+    using list_type =
+        Cabana::ParticleList<TEST_MEMSPACE, Cabana::Field::Position<3>, Foo,
+                             Cabana::Field::CommRank, Bar>;
+
+    list_type particles( "test_particles" );
+
+    // Resize the aosoa.
+    auto& aosoa = particles.aosoa();
+    std::size_t num_p = 10;
+    aosoa.resize( num_p );
+    EXPECT_EQ( particles.aosoa().size(), 10 );
+
+    // Populate fields.
+    auto px = particles.slice( Cabana::Field::Position<3>() );
+    auto pm = particles.slice( Foo() );
+    auto pc = particles.slice( Cabana::Field::CommRank() );
+    auto pf = particles.slice( Bar() );
+
+    Cabana::deep_copy( px, 1.23 );
+    Cabana::deep_copy( pm, 3.3 );
+    Cabana::deep_copy( pc, 5 );
+    Cabana::deep_copy( pf, -1.2 );
+
+    // Check the slices.
+    EXPECT_EQ( px.label(), "position" );
+    EXPECT_EQ( pm.label(), "foo" );
+    EXPECT_EQ( pc.label(), "comm_rank" );
+    EXPECT_EQ( pf.label(), "bar" );
+
+    // Check deep copy.
+    auto aosoa_host =
+        Cabana::create_mirror_view_and_copy( Kokkos::HostSpace(), aosoa );
+    auto px_h = Cabana::slice<0>( aosoa_host );
+    auto pm_h = Cabana::slice<1>( aosoa_host );
+    auto pc_h = Cabana::slice<2>( aosoa_host );
+    auto pf_h = Cabana::slice<3>( aosoa_host );
+    for ( std::size_t p = 0; p < num_p; ++p )
+    {
+        for ( int d = 0; d < 3; ++d )
+            EXPECT_DOUBLE_EQ( px_h( p, d ), 1.23 );
+
+        EXPECT_DOUBLE_EQ( pm_h( p ), 3.3 );
+
+        EXPECT_EQ( pc_h( p ), 5 );
+
+        for ( int i = 0; i < 3; ++i )
+            for ( int j = 0; j < 3; ++j )
+                EXPECT_DOUBLE_EQ( pf_h( p, i, j ), -1.2 );
+    }
+
+    // Locally modify.
+    Kokkos::parallel_for(
+        "modify", Kokkos::RangePolicy<TEST_EXECSPACE>( 0, num_p ),
+        KOKKOS_LAMBDA( const int p ) {
+            auto s = Cabana::Impl::Index<
```

Comment:
Definitely want a helper function for this

Reply:
I added, but maybe `access` isn't the best name